core/keychain/keychain.go

```diff
@@ -38,26 +44,43 @@ func New(store db.Store) *keychain {
 // If there's already a key pair stored, it returns an error.
 // Use GenerateKeyPairWithForce if you want to override existing keys
 func (kc *keychain) GenerateKeyPair() ([]byte, []byte, error) {
-	if val, _ := kc.store.Get([]byte(PublicKeyStoreKey)); val != nil {
+	return kc.GenerateNamedKeyPair(PrivateKeyStoreKey, PublicKeyStoreKey)
+}
+
+func (kc *keychain) GenerateRootKeyPair() ([]byte, []byte, error) {
+	return kc.GenerateNamedKeyPair(RootPrivateKeyStoreKey, RootPublicKeyStoreKey)
+}
+
+func (kc *keychain) GenerateNamedKeyPair(privkeystorekey string, pubkeystorekey string) ([]byte, []byte, error) {
```

Comment:
perhaps this should be a private function and not a public one? It looks more like an internal implementation details.